mtags/src/main/scala/scala/meta/internal/mtags/ScalaToplevelMtags.scala

```diff
@@ -290,12 +290,27 @@ class ScalaToplevelMtags(
           )
         case DEF | VAL | VAR | GIVEN
             if expectTemplate.map(!_.isExtension).getOrElse(true) =>
+          val isInParen =
+            region match {
+              case (_: Region.InParenCaseClass) | (_: Region.InParenClass) =>
+                true
+              case _ => false
+            }
+          val isImplicit =
+            (isInParen && expectTemplate.exists(
+              _.isImplicit
+            )) || (!isInParen && region.isImplicit)
           if (needEmitTermMember()) {
             withOwner(currRegion.termOwner) {
-              emitTerm(currRegion)
+              emitTerm(currRegion, isImplicit)
             }
           } else scanner.nextToken()
-          loop(indent, isAfterNewline = false, currRegion, newExpectIgnoreBody)
+          loop(
+            indent,
+            isAfterNewline = false,
+            currRegion,
+            if (isInParen) expectTemplate else newExpectIgnoreBody
```

Comment:
Why this change?

Reply:
```
implicit class A(val v: Int) {

}
```
So when we are on `class` we'll add something like expect implicit class body, and then in parenthesis we'll emit `val v` but we want to keep the information about the "expect implicit class body", it will be needed to construct correct region on `{`.

I wanted to write that this is safe, since there are no `{ }` inside the constructor definition, but I'm second guessing this because of `RefinedTypes`.

Reply:
Fixed for refined types.